parser/metadecoders/decoder.go

```diff
@@ -36,16 +36,22 @@ import (
 
 // Decoder provides some configuration options for the decoders.
 type Decoder struct {
-	// Delimiter is the field delimiter used in the CSV decoder. It defaults to ','.
+	// Delimiter is the field delimiter. Used in the CSV decoder. Default is
+	// ','.
 	Delimiter rune
 
-	// Comment, if not 0, is the comment character used in the CSV decoder. Lines beginning with the
-	// Comment character without preceding whitespace are ignored.
+	// Comment, if not 0, is the comment character. Lines beginning with the
+	// Comment character without preceding whitespace are ignored. Used in the
+	// CSV decoder.
 	Comment rune
 
 	// If true, a quote may appear in an unquoted field and a non-doubled quote
-	// may appear in a quoted field. It defaults to false.
+	// may appear in a quoted field. Used in the CSV decoder. Default is false.
 	LazyQuotes bool
+
+	// Whether to unmarshal to a map instead of an array. Used in the CSV
+	// decoder. Default is false.
+	ToMap bool
```

Comment:
I suggest we make this an enum, e.g. `targetType` (slice, map - default slice).

I'm pretty sure we at one point will get more options here (e.g. the new Go iterator types).

Reply:
I've made the change. Note that `targetType` is still a CSV-specific option.